### PR TITLE
[Build] Add linting to gate

### PR DESCRIPTION
**Pull request by riedgar-ms**
## Description

Add an extra job to the GitHub Action which forms our PR gate to also run flake8 on the codebase. This was done in the ADO pipelines, but is apparently not yet done in GitHub.

Fix some files which had line length violations.

## Tests

- [x] no new tests required
- [ ] new tests added
- [ ] existing tests adjusted

## Documentation

- [x] no documentation changes needed
- [ ] user guide added or updated
- [ ] API docs added or updated
- [ ] example notebook added or updated

### CODE REVIEW

**Review by adrinjalali — APPROVED** [review]
Happy to have this :)

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -60,3 +60,13 @@ jobs:
...
           verbose: true
+
+  linting:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v3
+        with:
+          python-version: 3.9
+      - run: pip install -r requirements-dev.txt
```
[the remark this answers is not in the document]
Comment (riedgar-ms) [reply .github/workflows/build-test.yml:71]: That's the file with our `flake8` requirements (and a bunch of other things, of course).

[the remark this answers is not in the document]
Comment (adrinjalali) [reply .github/workflows/build-test.yml:71]: I'd personally put those here I guess. But no hard feelings, my approval stands

[the remark this answers is not in the document]
Comment (riedgar-ms) [reply .github/workflows/build-test.yml:71]: I think it's best to have them in a requirements file, so that users can easily run the same linting locally.

**Pull request merged**